modules/nf-core/software/bwa/mem/functions.nf

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
These function files are the same for each module, right? Wouldn't it make sense to only have them once then? Sorry, if there was a discussion on this already and I missed it 🙈

Reply:
They are the same for now.
But I think the point of having them separate is that when you update a module, you update its `functions.nf` companion at the same time, so you can modules that are not all as up to date with the nf-core/modules repo.
But yeah, it's a discussion we had somewhere, and I think that's the consensus for now.
I assume one improvement could be to have a tagged version of this file, so that we use it only once per repo/version and not one per module

Reply:
Yeah, Harshil mentioned that these would be unnecessary once the newer version of `nf-core/tools` is released and can take care of deps common to all `nf-core/modules` such as the `functions.nf`. 

In the meantime, we toil with repetition 🚴